uiplib/scrape.py

```diff
@@ -32,33 +31,58 @@ def dlProgress(count, blockSize, totalSize):
     sys.stdout.flush()
 
 
-def get_images(url):
-    '''
-    scrape images from /r/wallpapers
-    '''
+'''scrape images'''
+def get_images(flag, url):
+
     soup = make_soup(url)
-    # this makes a list of bs4 element tags
-    thumbnails = soup.find_all("a", class_="thumbnail", href=True)
 
-    """Thumbnails in /r/wallpapers contain href to original
-    full-sized image."""
+    if flag == 1:       #For Unsplash
+        #Selects desired bs4 tags
+        a_tag = soup.select('.y5w1y .hduMF .tPMQE a')
+
+        image_links = []
+
+        if not a_tag:
+            print ('No matching image found')
+            return
+
+        for x in a_tag:
+            image_links.append(x['href'])
+            if(len(image_links) == NUMBER_OF_IMAGES_TO_PARSE):
+                break
 
-    image_links = []
-    if not thumbnails:
-        print('No matching image found')
-        return
+    elif flag == 2:       #For Reddit
 
-    for link in thumbnails:
-        if link['href'].endswith(('jpg', 'png', 'jpeg', 'bmp')):
-            image_links.append(link['href'])
-        if(len(image_links) == NUMBER_OF_IMAGES_TO_PARSE):
-            break
+        # this makes a list of bs4 element tags
+        thumbnails = soup.find_all("a", class_="thumbnail", href=True)
+
+        image_links = []
+
+        if not thumbnails:
+            print ('No matching image found')
+            return
+
+        for link in thumbnails:
+            #pattern matching to extract links
+            if link['href'].endswith(('jpg', 'png', 'jpeg', 'bmp')):
+                image_links.append(link['href'])
+            if(len(image_links) == NUMBER_OF_IMAGES_TO_PARSE):
+                break
 
     for image in image_links:
-        path = os.path.join(CURR_DIR, PICS_FOLDER)
+        path = os.path.join(CURR_DIR,PICS_FOLDER)
+
         if not os.path.exists(path):
             os.makedirs(path)
-        filename = image.split('/')[-1]
+
+        if flag == 1:       #unsplash
+            str1="http://unsplash.com/photos/"
+            str2="/download"
+            #Find text between two substrings
+            filename = image.split(str1)[1].split(str2)[0]+".jpg"
```

Comment:
```
filename = image.split('/')[-1] 
```
this line works generally for all image urls

 and generally do follow one line
```
Leave the campground cleaner than the way you found it.
```